### ISSUE TITLE
Clone specific version of the codebase from the GitHub

### ISSUE THREAD

**Issue by lexsys27** [poanetwork/deployment-playbooks#84]
Currenly git task downloads the latest version of code committed to GitHub repo.

```
git:
  repo: "https://github.com/{{ MAIN_REPO_FETCH }}/chain-explorer"
  dest: "{{ home }}/chain-explorer"
```

Proposal is to fix the version to a specific commit like this:

```
git:
  repo: "https://github.com/{{ MAIN_REPO_FETCH }}/chain-explorer"
  dest: "{{ home }}/chain-explorer"
  version: "acee07c"
```

The main reason is reproducible deployments. When I deploy from the same configuration I expect to get the same result. If I rely on the `latest` version of the code in the repository then my deployment in 1 month will be different from my deployment today.

This scenario is hard to troubleshoot. Also fixing the version allows controlled dependency upgrade that is visible in the commit history.

**Comment by phahulin** [comment]
> Currenly git task downloads the latest version of code committed to GitHub repo

this is usually what's intended - to have the latest version when deploying new instance

**Comment by lexsys27** [comment]
`ansible-lint` has following rule:

```
ANSIBLE0004: Git checkouts must contain explicit version
  All version control checkouts must point to an explicit commit or tag, not just "latest"
```

**Comment by phahulin** [comment]
Can we disable this rule?
Also, this might be a common ground: introduce new variables like `explorer_commit`, `netstats_commit`, set them to `"latest"` for existing setup (in `all.example` and `all.network` files) and add `version` to the tasks where necessary. This way I think linter will be happy.

**Comment by lexsys27** [comment]
Technically this check can be skipped but then the deployment becomes non-repeatable.

The repeatability of the deployment is important in following scenarios:

One deploys `staging` environment and runs some custom tests for one week. It may be QA, performance testing, integration with some other system or longevity testing. After tests pass he deploys to `prod`. When using the `latest` tag there is no guarantee that he deploys the same version he tested in `staging` - so invalidating the whole purpose of such testing.

The same applies to the case when company seeks the security audit and wants to deploy in `prod` exactly the same version that was audited.

In my experience sometimes one needs to re-create customer environment while investigating the bug. If the deployment is non-repeatable and bug doesn't show up in the recent version it nearly impossible to fix it for that particular installation.

I would suggest adding release versioning for the parts of POA Network and targeting the specific release from the playbook. Release may be a tag or a release branch. Version number is stored in the role-scoped variable.

```
explorer_version = "0.1.2"

git:
  repo: "https://github.com/{{ MAIN_REPO_FETCH }}/chain-explorer"
  dest: "{{ home }}/chain-explorer"
  version: "{{ explorer_version }}"
```

**Comment by phahulin** [comment]
These are all valid points, thank you, I agree.

With regard to releases, I understand `version` accepts `HEAD`, commit sha, release, branch name, so we can use commit shas or HEAD for now. Could you do introduce all those variables as part of this issue?

**Comment by lexsys27** [comment]
Sure! See the latest commits to the `ansible-lint` branch

**Comment by lexsys27** [comment]
Closed as implemented and merged in #82
